c10/core/TensorImpl.h

```diff
@@ -3169,7 +3169,7 @@ class C10_TensorImpl_Size_Check_Dummy_Class : private TensorImpl {
 #if UINTPTR_MAX == 0xFFFFFFFF
   // This is a 32-bit system
   static constexpr bool check_sizes() {
-    constexpr size_t tsize = 20 * sizeof(int64_t);
```

Comment:
That sounds like too much? Is bad padding happening?

Reply:
It's probably related  to the `optional` aligning in a way that requires `3 * sizeof(widest type in Device)` bytes of storage. That wasn't a problem before as both device type and index were `int8_t`.